drwmutex.go

```diff
@@ -122,11 +136,15 @@ func (dm *DRWMutex) lockBlocking(isReadLock bool) {
 				copy(dm.writeLocks, locks[:])
 			}
 
-			return
+			return true
+		}
+		if time.Since(start) >= timeout { // Are we past the timeout?
```

Comment:
I see retry channel is not user controllable ie this method creates and uses it. Is it good idea to work with `doneCh` over `timeout` from api consumer?

If not, I would prefer different name than `timeout` and that needs to be properly documented.